pyluna-pathology/luna/pathology/cli/generate_tile_labels.py

```diff
@@ -1,103 +1,129 @@
+        d_collections[label] = GeometryCollection( d_collections[label] )
 
-    Returns:
-        None
-    """
-    logger = logging.getLogger(f"[datastore={datastore_id}]")
-
-    # Do some setup
-    cfg = ConfigSet("APP_CFG", config_file=app_config)
-    datastore   = DataStore_v2(method_data.get("root_path"))
-    method_id   = method_data.get("job_tag", "none")
-
-    image_path  = datastore.get(datastore_id, method_data['input_wsi_tag'], "WholeSlideImage")
-    logger.info(f"Whole slide image path: {image_path}")
-
-    # get image_id
-    # TODO - allow -s to take in slide (datastore_id) id
-    image_id = datastore_id
-
-    try:
-        if image_path is None:
-            raise ValueError("Image node not found")
-
-        # Data just goes under namespace/name
-        # TODO: This path is really not great, but works for now
-        output_dir = os.path.join(method_data.get("root_path"), datastore_id, method_id, "TileImages", "data")
-        if not os.path.exists(output_dir): os.makedirs(output_dir)
-
-        logger.info(f"Writing to output dir: {output_dir}")
-        # properties = pretile_scoring(image_path, output_dir, method_data.get("annotation_table_path"), method_data, image_id)
-        properties = {}
-    except Exception as e:
-        logger.exception (f"{e}, stopping job execution...")
-        raise e
-
-    with open(os.path.join(output_dir, "metadata.json"), "w") as fp:
-        json.dump(properties, fp)
+    df_tiles = pd.read_csv(input_slide_tiles).set_index('address')
+    l_regional_labels = []
+    l_intersection_areas = []
+
+    for _, row in tqdm(df_tiles.iterrows(), total=len(df_tiles)):
+        tile_x, tile_y, tile_extent = row.x_coord, row.y_coord, row.xy_extent
+
+        tile_polygon = Polygon([
+            (tile_x,               tile_y),
+            (tile_x,               tile_y+tile_extent),
+            (tile_x+tile_extent,   tile_y+tile_extent),
+            (tile_x+tile_extent,   tile_y),
+        ])
+
+        tile_label = None
+        max_overlap = 0.0
+        for label in d_collections.keys(): 
+            intersection_area = d_collections[label].intersection(tile_polygon).area / tile_polygon.area
+            if intersection_area > max_overlap:
+                tile_label, max_overlap = label, intersection_area
+
+        logger.info(tile_polygon)
+        l_regional_labels.append(tile_label)
+        l_intersection_areas.append(max_overlap)
+
+    df_tiles['regional_label']    = l_regional_labels
+    df_tiles['intersection_area'] = l_intersection_areas
+
+    logger.info ( df_tiles.loc[df_tiles.intersection_area > 0] )
+
+    output_header_file = f"{output_dir}/{slide_id}.regional_label.tiles.csv"
+    df_tiles.to_csv(output_header_file)        
+    
+    properties = {
+        "slide_tiles": output_header_file, # "Tiles" are the metadata that describe them
+        "segment_keys": {'regional_annotation_id': f"dsa-{collection_name}-{annotation_name}"}
```

Comment:
could we use annotation_name or annotation id here? i'd remove the hard-coded "dsa-" here to support annotations generated by other means.